omniscidb/QueryEngine/QueryTemplateGenerator.cpp

```diff
@@ -1019,26 +1011,25 @@ class NonGroupedQueryTemplateGenerator : public QueryTemplateGenerator {
                                  "",
                                  bb_exit);
         } else {
+          // TODO(adb): (1) need to set address space on this opaque pointer properly
+          //            (2) need to make sure this sequence is correct w/r/t opague
+          //            pointer types
           auto out_gep = llvm::GetElementPtrInst::CreateInBounds(
-              output_buffers->getType()->getPointerElementType(),
-              output_buffers,
-              col_idx,
+              get_int_type(64, mod->getContext()), output_buffers, col_idx, "", bb_exit);
+          auto col_buffer = new llvm::LoadInst(
+              llvm::PointerType::get(mod->getContext(), /*AddressSpace=*/0),
```

Comment:
Input column buffer, as well as output buffer pointers, have global (1) address space at the topmost level. The wrapper casts them to generic (4, this matches our local type in the codegen traits; I should rename it to "generic" for clarity) and calls the query stub. GEP creation does not require explicit address space mentioning though, you can use `getType()`, here's an example: https://github.com/kurapov-peter/L0Snippets/blob/main/common/helpers.cpp#L46